### PR TITLE
Cleaning up cb.vp in preparation for writing a magma version

### PR DESCRIPTION
**Pull request by dillonhuff**
@alexcarsello I'm cleaning up the genesis code for cb.vp. I removed a superfluous modulo from feedthrough calculation in cb.vp and deleted a wire  ```_genesis_wire...``` that seemed to never be used.

Do these changes look ok to you?

I think the next step is to add a magma generator for the cb and regression test it against the current generator. I'm working on a simple script to do that now.

### CODE REVIEW

**Comment by dillonhuff** [comment]
I also just added a little test bench and python script to run it. That test is not part of the flow, so it should not affect anything. It is just there so that when I start writing the magma generator I can have something to test against.

**Comment by alexcarsello** [comment]
I know there is a reason for the __genesis_wire, but I do not know what it is. Something about uniquification. I'm sure whatever it is should be done in a different way, and wouldn't be necessary in Magma anyway.

**Comment by dillonhuff** [comment]
Ok. I'll reintroduce the genesis_wire for now.

**Comment by dillonhuff** [comment]
Just added it back

File: hardware/generator_z/cb/cb.vp [hunk 1]
```diff
@@ -6,10 +6,12 @@
 //; my $default_value = parameter(Name=>'default_value', val=> 0, doc=>'set default cb output value on reset. has_constant should be 1 to enable this feature'); 
 //; my @feedthrough = split('',$feedthrough_outputs);
 
+
+
 module `mname` (
 clk, reset,
 //; for(my $i=0; $i<$num_tracks; $i++) {
-//;  if ($feedthrough[($i%$num_tracks)]==1) {
```
Comment (rsetaluri) [inline hardware/generator_z/cb/cb.vp]: Do we need both `feedthrough_outputs` and `num_tracks`? The description of `feedthrough_outputs` mentions that the bit-length of `feedthrough_outputs` is equal to num_tracks.

Also we can probably make analogous changes on lines 44 and 108.

File: hardware/generator_z/cb/cb.vp [hunk 2]
```diff
@@ -33,8 +35,9 @@ read_data
 
   /* verilator lint_off UNUSED */
   input [31:0] config_addr;
-  /* verilator lint_on UNUSED */
-  wire ___genesis_wire_`${default_value}`_`${width}`_`${num_tracks}`_`${feedthrough_outputs}`_`${has_constant}`;
+
+  /* verilator lint_off UNUSED */
```
Comment (rsetaluri) [inline hardware/generator_z/cb/cb.vp]: why are these moved around?

**Comment by rsetaluri** [comment]
Ignore the approval/dismissal of the review :). But my original comments are valid, and I addressed them in the latest change.

**Comment by leonardt** [comment]
I like the addition of the test!

File: hardware/generator_z/cb/cb.vp [hunk 3]
```diff
@@ -1,15 +1,21 @@
 //; use POSIX;
 //; my $width  = parameter(Name=>'width', val=> 16, doc=>'Bus width for CB');
+//; # num_tracks is a redundant parameter and is unused!! It has been replaced
+//; # with the bit-length of feedthrough_outputs.
 //; my $num_tracks = parameter(Name=>'num_tracks', val=> 10, doc=>'num of tracks for CB');
```
[the remark this answers is not in the document]
Comment (rsetaluri) [reply hardware/generator_z/cb/cb.vp]: It will likely break things since consumers of this generator expect that parameter to be present. Ideally we would rip that out, but for the sake of keeping this change small I kept the `num_tracks` param.

**Comment by steveri** [comment]
@dillonhuff @rsetaluri @leonardt Here is a very old git pull request that I found, still outstanding.  Can anyone tell me what is its status..should we merge or close?  I notice the checks are currently failing...

**Comment by leonardt** [comment]
I would say close. This was a precursor to the garnet effort, but I think at this point it's safe to assume any garnet specific code should be over on that repository.

**Comment by leonardt** [comment]
Woops, accidentally hit close and comment instead of comment, but I would vote to close anyways :)
